File: GameLayers/attackLayer.js
import { config } from '../config.js'


export default class UnderCostBudgetAttackLayer {
    execute = function(world) {
        var cards = world.attackCards
        for(var i = 0; i < cards.length; i++){
            var card = cards[i]
            if(card.cost <= world.turnBudget){
                world.boss.health = world.hero.attackAction(world.boss, card)
            }
            world.turnBudget -= card.cost
        }
    }
}

File: python/GameLayers/attackLayer.py
import math
import config

class UnderCostBudgetAttackLayer:
    def execute(self, world):
        cards = world.attackCards
        for card in cards:
            if card.cost <= world.turnBudget:
                world.boss.health = world.hero.attackAction(world.boss, card)
            world.turnBudget -= max(1, card.cost)
        
class DefaultAttackLayer:
    def execute(self, world):
        cards = world.attackCards
        for card in cards:
            world.boss.health = world.hero.attackAction(world.boss, card)
        world.turnBudget -= max(1, card.cost)
